src/modules/ekf2/EKF/yaw_estimator/EKFGSF_yaw.cpp

```diff
@@ -367,20 +350,10 @@ bool EKFGSF_yaw::updateEKF(const uint8_t model_index, const Vector2f &vel_NE, co
 	_ekf_gsf[model_index].X.xy() += delta_state.xy();
 	_ekf_gsf[model_index].X(2) = wrap_pi(_ekf_gsf[model_index].X(2) + yawDelta);
 
-	// apply the change in yaw angle to the AHRS
-	// take advantage of sparseness in the yaw rotation matrix
-	const float cosYaw = cosf(yawDelta);
-	const float sinYaw = sinf(yawDelta);
-	const float R_prev00 = _ahrs_ekf_gsf[model_index].R(0, 0);
-	const float R_prev01 = _ahrs_ekf_gsf[model_index].R(0, 1);
-	const float R_prev02 = _ahrs_ekf_gsf[model_index].R(0, 2);
-
-	_ahrs_ekf_gsf[model_index].R(0, 0) = R_prev00 * cosYaw - _ahrs_ekf_gsf[model_index].R(1, 0) * sinYaw;
-	_ahrs_ekf_gsf[model_index].R(0, 1) = R_prev01 * cosYaw - _ahrs_ekf_gsf[model_index].R(1, 1) * sinYaw;
-	_ahrs_ekf_gsf[model_index].R(0, 2) = R_prev02 * cosYaw - _ahrs_ekf_gsf[model_index].R(1, 2) * sinYaw;
-	_ahrs_ekf_gsf[model_index].R(1, 0) = R_prev00 * sinYaw + _ahrs_ekf_gsf[model_index].R(1, 0) * cosYaw;
-	_ahrs_ekf_gsf[model_index].R(1, 1) = R_prev01 * sinYaw + _ahrs_ekf_gsf[model_index].R(1, 1) * cosYaw;
-	_ahrs_ekf_gsf[model_index].R(1, 2) = R_prev02 * sinYaw + _ahrs_ekf_gsf[model_index].R(1, 2) * cosYaw;
+	// Apply the change in yaw angle to the AHRS using left multiplication to rotate
+	// the attitude around the earth Down axis
+	const Quatf dq(cosf(yawDelta / 2.f), 0.f, 0.f, sinf(yawDelta / 2.f));
+	_ahrs_ekf_gsf[model_index].q = (dq * _ahrs_ekf_gsf[model_index].q).normalized();
```

Comment:
Just double checking about the order - to me it makes more sense to first apply the yaw and then the remaining tilt as the rotation is applied from right to left. Could you confirm?

Reply:
Rotating a vector is from right to left, yes, but here we're simply chaining rotations here (from left to right). So yes, first rotate around yaw and then tilt.

Reply:
ah okay, because in the Quaternion header file it says the order is from right to left. But if it's from left to right then makes sense!